core/src/data/rasterSource.cpp

```diff
@@ -10,10 +10,86 @@
 
 namespace Tangram {
 
+class RasterTileTask : public DownloadTileTask {
+public:
+    RasterTileTask(TileID& _tileId, std::shared_ptr<DataSource> _source, bool _subTask)
+        : DownloadTileTask(_tileId, _source, _subTask) {}
+
+    std::shared_ptr<Texture> m_texture;
+
+    bool hasData() const override {
+        return bool(rawTileData) || bool(m_texture);
+    }
+
+    bool isReady() const override {
+        if (m_subTask) {
+            return bool(m_texture);
+        } else {
+            return bool(m_tile);
+        }
+    }
+
+    void onProcess(TileBuilder& _tileBuilder) override {
+
+        auto source = reinterpret_cast<RasterSource*>(m_source.get());
+
+        if (!m_texture) {
+            // Decode texture data
+            m_texture = source->createTexture(*rawTileData);
```

Comment:
can we put in an assert here that m_texture was never set before!

But, I do not think there is no code path currently which will make onProcess be called twice for the same RasterTask!